code/_onclick/item_attack.dm

```diff
@@ -71,15 +71,21 @@ avoid code duplication. This includes items that may sometimes act as a standard
 
 //I would prefer to rename this attack_as_weapon(), but that would involve touching hundreds of files.
 /obj/item/proc/attack(mob/living/M, mob/living/user, var/target_zone = BP_CHEST)
-
 	if(!force || (flags & NOBLUDGEON))
 		return 0
+
 	if(M == user && user.a_intent != I_HURT)
 		return 0
 
 	if(user.is_pacified())
 		return 0
 
+	if(issilicon(user))
```

Comment:
I would make it `/mob/proc/allow_attack` and override for silicons & robots. Then you don't need this check or to typecast.

You can make `allow_attack()` return the appropriate message to put in the warning.